File: leetcode/2206-divide-array-into-equal-pairs/solution.cpp
#include <vector>
#include <unordered_map>
using namespace std;

class Solution {
public:
    bool divideArray(vector<int>& nums) {
        unordered_map<int, int> counts;
        for (int num : nums) counts[num]++;

        for (auto& [value, c] : counts) {
            if (c % 2 != 0) return false;
        }
        return true;
    }
};

File: leetcode/2206-divide-array-into-equal-pairs/Solution.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public boolean divideArray(int[] nums) {
        Map<Integer, Integer> counts = new HashMap<>();
        for (int num : nums) {
            counts.merge(num, 1, Integer::sum);
        }

        for (int c : counts.values()) {
            if (c % 2 != 0) return false;
        }
        return true;
    }
}
